src/client/Microsoft.Identity.Client/ApiConfig/Executors/ConfidentialClientExecutor.cs

```diff
@@ -59,7 +62,13 @@ public ConfidentialClientExecutor(IServiceBundle serviceBundle, ConfidentialClie
                 _confidentialClientApplication.AppTokenCacheInternal).ConfigureAwait(false);
        
             requestParams.SendX5C = clientParameters.SendX5C ?? false;
-            
+
+            if (ServiceBundle.Config.UseManagedIdentity)
+            {
+                ManagedIdentityClient managedIdentityClient = new ManagedIdentityClient(requestContext);
+                ServiceBundle.Config.AppTokenProvider = managedIdentityClient.AppTokenProviderImplAsync;
+            }
+
             var handler = new ClientCredentialRequest(
```

Comment:
In `ClientCredentialRequest` we log an error if the tenant is "common" - 

<img width="640" alt="image" src="https://user-images.githubusercontent.com/12273384/204105743-58772acb-0ffd-4f5f-98df-19a10adfea89.png">


this should be disabled.